node/src/reactor.rs

```diff
@@ -333,7 +342,17 @@ where
         trace!(?event, ?q);
 
         // Dispatch the event, then execute the resulting effect.
+        let start = self.clock.start();
         let effects = self.reactor.dispatch_event(effect_builder, rng, event);
+        let end = self.clock.end();
+        let delta = self.clock.delta(start, end);
+
+        if delta > DISPATCH_EVENT_THRESHOLD {
+            warn!(
+                ns = delta.into_nanos(),
+                "previous event took very long to dispatch"
```

Comment:
We should log what type of event it was.

Reply:
I know you can take the previous `trace!` but there may be a lot of other log lines in-between those.